hub/core/dataset/dataset.py

```diff
@@ -799,6 +822,45 @@ def tensors(self) -> Dict[str, Tensor]:
             for t in self._all_tensors_filtered
         }
 
+    @property
+    def branches(self):
+        """Lists all the branches of the dataset.
+        Returns:
+            List of branches.
+        """
+        return list(self.version_state["branch_commit_map"])
+
+    @property
+    def commits(self) -> List[Dict]:
+        """Lists all the commits leading to the current dataset state.
+        Returns:
+            List of dictionaries containing commit information.
+        """
+        commits = []
+        commit_node = self.version_state["commit_node"]
+        while commit_node:
+            if commit_node.commit_time is not None:
```

Comment:
In what scenario is `commit_time == None`?  What is being checked here?

Reply:
commit_time is None when the commit hasn't been made yet i.e. we're on the head commit

Reply:
For better readability, maybe:

``` python
is_head_commit = lambda node: node.commit_time is None

...
    if not is_head_commit(commit_node):
        ...
```


```